finetune/base.py

```diff
@@ -356,30 +476,91 @@ def _cached_inference(self, Xs, mode=None):
 
         return predictions
 
+
+    def separate_cached_inference(self,Xs,mode=None):
+        features = None
+        #self.get_separate_estimators()
+        n = len(self._data)
+        hooks =self.hooks
+        input_fn = self.input_pipeline.get_predict_input_fn(self._data_generator)
+        #self.get_separate_estimators()
+        if self._predictions is None:
+            self._predictions =  self.featurizer_est.predict(
+                    input_fn=input_fn, predict_keys=None, hooks=[hooks[0]])
+            hooks[0].need_to_refresh = False
```

Comment:
If we are enforcing an order of a list like this then this should probably be a named tuple.